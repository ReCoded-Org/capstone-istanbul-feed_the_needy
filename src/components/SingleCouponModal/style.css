.ant-card-meta-title {
  text-align: center;
  font-weight: bold;
  font-size: -webkit-xxx-large;
}

.singleCouponContent {
  padding-top: inherit;
  text-align: center;
}

.ant-btn.ant-btn-primary {
  background-color: #e17007;
}
