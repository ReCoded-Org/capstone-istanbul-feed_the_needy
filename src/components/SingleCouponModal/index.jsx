import React, { useEffect, useState } from "react";
import "./style.css";
import { Modal, Card, Typography, Button } from "antd";
import { useTranslation } from "react-i18next";
import { LinkOutlined } from "@ant-design/icons";
import firebase from "../../firebaseConfig";
import defaultRestaurantLogo from "../../images/defaultRestaurantLogo.jpg";

const db = firebase.firestore();
const { Text, Title } = Typography;
const { Meta } = Card;

const SingleCouponModal = ({
  compName,
  price,
  visibleModal,
  setVisibleModal,
}) => {
  const { t } = useTranslation();
  const [coupon, setCoupon] = useState([]);
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    const unsubscribe = db
      .collection("restaurants")
      .where("name", "==", compName || null) // this can changed with the restaurant id later
      .onSnapshot((snapshot) => {
        const dataArr = [];
        snapshot.forEach((doc) => {
          dataArr.push({ ...doc.data(), docId: doc.id });
        });
        setCoupon(dataArr[0]);
        setLoading(false);
      });
    return unsubscribe;
  }, [compName]);

  const handleOk = () => {
    setVisibleModal(false);
  };

  const handleCancel = () => {
    setVisibleModal(false);
  };

  return (
    <div className="singleCouponModal">
      <Modal
        title={t("singleCouponModal.title")}
        visible={visibleModal}
        onOk={handleOk}
        onCancel={handleCancel}
        footer={[
          <Button type="primary" key="ok" onClick={handleOk}>
            {t("singleCouponModal.OK")}
          </Button>,
        ]}
      >
        <Card
          loading={loading}
          bordered={false}
          style={{ width: "100%", height: "96.5%" }}
          cover={
            <img
              style={{ maxHeight: "21rem" }}
              alt={t("singleCouponModal.alt")}
              src={coupon.logoURL || defaultRestaurantLogo}
            />
          }
        >
          <Meta title={`${price}â‚º`} />
          <div className="singleCouponContent">
            <Title level={3}>{coupon.name}</Title>
            <a
              title={t("editOrganization.exampleCoupon.link")}
              target="_blank"
              rel="noopener noreferrer"
              href={coupon.website}
            >
              <LinkOutlined />
            </a>
            <br />
            <Text>{coupon.description}</Text>
          </div>
        </Card>
      </Modal>
    </div>
  );
};

export default SingleCouponModal;
