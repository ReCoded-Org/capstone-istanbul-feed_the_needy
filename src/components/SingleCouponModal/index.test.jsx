import React from "react";
import renderer from "react-test-renderer";
import SingleCouponModal from "./index";

jest.mock("react-i18next", () => ({
  useTranslation: () => ({ t: (key) => key }),
}));

it("SingleCouponModal renders correctly", () => {
  const tree = renderer.create(<SingleCouponModal />).toJSON();
  expect(tree).toMatchSnapshot();
});
