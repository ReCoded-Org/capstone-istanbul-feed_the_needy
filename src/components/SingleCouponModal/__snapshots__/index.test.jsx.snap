// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleCouponModal renders correctly 1`] = `
<div
  className="singleCouponModal"
/>
`;
